source/GarminSDComms.mc

```diff
@@ -217,6 +216,7 @@ class GarminSDComms {
     }
     lastOnReceiveResponse = responseCode;
     lastOnReceiveData = data;
+    mDataRequestInProgress = 0;
```

Comment:
Moving this re-setting of the flag to the end of the function means it will block sending data until we have sent settings and requested the status.  Now this will prevent overloading which is good, but sending data is the critical part that we want to happen at regular intervals.    Let's try it and see how it goes - the thing to watch for is how often the clock on the phone app updates - it should be every 5 seconds - as long as that is the case, then this is the right thing to do.